core/domain/question_domain.py

```diff
@@ -229,3 +237,64 @@ def update_question_data(self, question_data):
             question_data: dict. A dict representing the question data.
         """
         self.question_data = question_data
+
+    def add_skill(self, skill_name, user_id):
```

Comment:
Sorry for not catching this earlier, but ... why is this here? Domain object files should not depend on services. They should just be a set of plain Python objects.

Also this seems to be misnamed, it is adding a question ID to a skill in the collection. Shouldn't this be part of collection_services or perhaps question_services?

Similar comments apply to remove_skill.

Reply:
Updated, for the naming part, I think I followed the doc. Should I rename it to add_question_id_to_skill?

Reply:
Yeah, that sounds fine to me. Sorry about the doc!

Reply:
Updated.